.env.example

```diff
@@ -1,4 +0,0 @@
-ALGOLIA_API_KEY=
-ALGOLIA_APP_ID=
-NPM_TOKEN=
-SITE_URL=https://thegraph.com/docs
```

Comment:
This file was redundant because we already have `.env` checked in the repo. It's weird but [that's what Next recommends](https://nextjs.org/docs/basic-features/environment-variables#default-environment-variables). The `.env` file basically serves as `.env.example` but with some default values (the ones that aren't secrets), and it can be duplicated into `.env.local` to add secrets.